cmd/cstor-pool-mgmt/controller/backup-controller/handler.go

```diff
@@ -0,0 +1,237 @@
+		return "", nil
+	}
+	return string(apis.BKPCStorStatusInvalid), nil
+}
+
+// addEventHandler will change the state of backup to Init state.
+func (c *BackupController) addEventHandler(bkp *apis.BackupCStor) (string, error) {
+	if !IsPendingStatus(bkp) {
+		return string(apis.BKPCStorStatusInvalid), nil
+	}
+	return string(apis.BKPCStorStatusInit), nil
+}
+
+// syncEventHandler will perform the backup if a given backup is in init state
+func (c *BackupController) syncEventHandler(bkp *apis.BackupCStor) (string, error) {
+	// If the backup is in init state then only we will complete the backup
+	if IsInitStatus(bkp) {
+		bkp.Status = apis.BKPCStorStatusInProgress
+		_, err := c.clientset.OpenebsV1alpha1().BackupCStors(bkp.Namespace).Update(bkp)
+		if err != nil {
+			glog.Errorf("Failed to update backup:%s status : %v", bkp.Name, err.Error())
+			return "", err
+		}
+
+		err = volumereplica.CreateVolumeBackup(bkp)
+		if err != nil {
+			c.recorder.Event(bkp, corev1.EventTypeNormal, string(common.SuccessCreated), string(common.MessageResourceCreated))
+			glog.Errorf("Failed to create backup(%v): %v", bkp.ObjectMeta.Name, err.Error())
+			return string(apis.BKPCStorStatusFailed), err
+		}
+
+		c.recorder.Event(bkp, corev1.EventTypeNormal, string(common.SuccessCreated), string(common.MessageResourceCreated))
+		glog.Infof("backup creation successful: %v, %v", bkp.ObjectMeta.Name, string(bkp.GetUID()))
+		err = c.updateBackupCStorLast(bkp)
+		if err != nil {
+			return string(apis.BKPCStorStatusFailed), err
+		}
+		return string(apis.BKPCStorStatusDone), nil
+	}
+	return "", nil
+}
+
+// getBackupCStorResource returns a backup object corresponding to the resource key
+func (c *BackupController) getBackupCStorResource(key string) (*apis.BackupCStor, error) {
+	// Convert the key(namespace/name) string into a distinct name
+	glog.Infof("Finding backup for key:%s", key)
+	ns, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil, nil
+	}
+
+	bkp, err := c.clientset.OpenebsV1alpha1().BackupCStors(ns).Get(name, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			runtime.HandleError(fmt.Errorf("bkp '%s' in work queue no longer exists", key))
+			return nil, nil
+		}
+		return nil, err
+	}
+	return bkp, nil
+}
+
+// IsPendingStatus is to check if the backup is in a pending state.
+func IsPendingStatus(bkp *apis.BackupCStor) bool {
+	if string(bkp.Status) == string(apis.BKPCStorStatusPending) {
+		return true
+	}
+	return false
+}
+
+// IsInProgressStatus is to check if the backup is in in-progress state.
+func IsInProgressStatus(bkp *apis.BackupCStor) bool {
+	if string(bkp.Status) == string(apis.BKPCStorStatusInProgress) {
+		return true
+	}
+	return false
+}
+
+// IsInitStatus is to check if the backup is in init state.
+func IsInitStatus(bkp *apis.BackupCStor) bool {
+	if string(bkp.Status) == string(apis.BKPCStorStatusInit) {
```

Comment:
Same comments as above.
Please mark a TODO to rewrite this logic into predicate as per idiomatic maya standards